desktop/libs/notebook/src/notebook/connectors/hiveserver2.py

```diff
@@ -316,11 +317,11 @@ def execute(self, notebook, snippet):
     db = self._get_db(snippet, interpreter=self.interpreter)
 
     statement = self._get_current_statement(notebook, snippet)
-    session = self._get_session(notebook, snippet['type'])
+    compute = snippet.get('compute')
```

Comment:
Snippet will always have `compute`? And compute will have `name`?

Reply:
It can potentially be undefined in the UI: https://github.com/cloudera/hue/blob/f051be86f7647edb11bc53200a760b1449a31aad/desktop/core/src/desktop/js/apps/notebook/snippet.js#L288

Not sure in which scenario but perhaps when loading back old saved editor documents, from before computes were introduced. I'm guessing `is_compute` takes care of checking it @amitsrivastava?

Reply:
Now made the checks safer.